### ISSUE TITLE
Architecture: setup a central repository of all software components used in Trinity

### ISSUE THREAD

**Issue by hansthen** [clustervision/trinity#122]
We have had some problems in the past when new software components were introduced for a system. I propose to solve this by creating a centralised repository clone. This repository can keep track of all software packages that we need to a trinity installations.

**Comment by hansthen** [comment]
Is now inside the master_installer branch. Use trinity/master/create_repo_snapshot to create (and update) the rpm repository on master. repo snapshots are created under /install/post/<date>/, with /install/post/otherpgks pointing to the most current version.

**Comment by hansthen** [comment]
Still in master_installer branch. Will merge this into M4.

**Comment by themisa** [comment]
note: unplug master from internet and do installation. Currently docker centos image is still downloaded from internet. Need locally. 
Also update Dockerfiles not to use centos:latest, but a specific version.

**Comment by hansthen** [comment]
Let's split the requirement that docker image does not need to be installed from the internet. I think that if we have all the rpms and cds stable, most of our problems will go away.  Reassigning to Themis for testing of the master_installer branch. Once that is done, we will merge into M4.

**Comment by themisa** [comment]
@hansthen, @abhimukh Another thing that needs to be done locally: possibly firefox, and most importantly the image + requirements for the virtual login nodes. 
Docker should be another requirement? 

So i can confirm that in dtap both the xcat + openstack are deployed from local repos. How do we want to proceed?

**Comment by hansthen** [comment]
Just close it.

**Comment by themisa** [comment]
Ok. I will open a new enhancement request for login nodes.
